kafka-connect/src/main/java/io/tabular/iceberg/connect/channel/Coordinator.java

```diff
@@ -215,16 +220,38 @@ private void commitToTable(
       LOG.info("Nothing to commit to table {}, skipping", tableIdentifier);
     } else {
       if (deleteFiles.isEmpty()) {
-        AppendFiles appendOp = table.newAppend();
-        branch.ifPresent(appendOp::toBranch);
-        appendOp.set(snapshotOffsetsProp, offsetsJson);
-        appendOp.set(COMMIT_ID_SNAPSHOT_PROP, commitState.currentCommitId().toString());
-        if (vtts != null) {
-          appendOp.set(VTTS_SNAPSHOT_PROP, Long.toString(vtts));
-        }
-        dataFiles.forEach(appendOp::appendFile);
-        appendOp.commit();
+        Map<Integer, List<DataFile>> dataFilesBySpec = Maps.newHashMap();
+        dataFiles.forEach(
+            dataFile -> {
+              List<DataFile> list =
+                  dataFilesBySpec.computeIfAbsent(
+                      dataFile.specId(), ignored -> Lists.newArrayList());
+              list.add(dataFile);
+            });
+        Integer maxSpecId = dataFilesBySpec.keySet().stream().max(Integer::compare).get();
+
+        Transaction transaction = table.newTransaction();
+        dataFilesBySpec.entrySet().stream()
+            .sorted(Comparator.comparingInt(Map.Entry::getKey))
```

Comment:
I don't think sorting by spec is required here?

Reply:
Yea, I agree, it wasn't necessary, removed.